demos/smart_classroom_demo_gapi/main.cpp

```diff
@@ -1,700 +1,169 @@
-            double reid_threshold,
-            int min_size_fr,
-            bool crop_gallery,
-            bool greedy_reid_matching
-    )
-        : landmarks_detector(landmarks_detector_config),
-          face_reid(reid_config),
-          face_gallery(face_gallery_path, reid_threshold, min_size_fr, crop_gallery,
-                       face_registration_det_config, landmarks_detector, face_reid,
-                       greedy_reid_matching)
-    {
-        if (face_gallery.size() == 0) {
-            slog::warn << "Face reid gallery is empty!" << slog::endl;
-        } else {
-            slog::info << "Face reid gallery size: " << face_gallery.size() << slog::endl;
-        }
-    }
-
-    bool LabelExists(const std::string &label) const override {
-        return face_gallery.LabelExists(label);
-    }
-
-    std::string GetLabelByID(int id) const override {
-        return face_gallery.GetLabelByID(id);
-    }
-
-    std::vector<std::string> GetIDToLabelMap() const override {
-        return face_gallery.GetIDToLabelMap();
-    }
-
-    std::vector<int> Recognize(const cv::Mat& frame, const detection::DetectedObjects& faces) override {
-        std::vector<cv::Mat> face_rois;
-
-        for (const auto& face : faces) {
-            face_rois.push_back(frame(face.rect));
-        }
-
-        std::vector<cv::Mat> landmarks, embeddings;
-
-        landmarks_detector.Compute(face_rois, &landmarks, cv::Size(2, 5));
-        AlignFaces(&face_rois, &landmarks);
-        face_reid.Compute(face_rois, &embeddings);
-        return face_gallery.GetIDsByEmbeddings(embeddings);
-    }
-
-    void PrintPerformanceCounts(
-            const std::string &landmarks_device, const std::string &reid_device) override {
-        landmarks_detector.PrintPerformanceCounts(landmarks_device);
-        face_reid.PrintPerformanceCounts(reid_device);
-    }
-
-private:
-    VectorCNN landmarks_detector;
-    VectorCNN face_reid;
-    EmbeddingsGallery face_gallery;
-};
+#include "initialize.hpp"
+#include "stream_source.hpp"
 
 bool ParseAndCheckCommandLine(int argc, char *argv[]) {
-    // ---------------------------Parsing and validation of input args--------------------------------------
-
     gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
     if (FLAGS_h) {
         showUsage();
         showAvailableDevices();
         return false;
     }
-
     slog::info << "Parsing input parameters" << slog::endl;
-
     if (FLAGS_i.empty()) {
         throw std::logic_error("Parameter -i is not set");
     }
     if (FLAGS_m_act.empty() && FLAGS_m_fd.empty()) {
         throw std::logic_error("At least one parameter -m_act or -m_fd must be set");
     }
-
     return true;
 }
 
-}  // namespace
+std::string GetBinPath(const std::string &pathXML) {
+    CV_Assert(pathXML.substr(pathXML.size() - 4, pathXML.size()) == ".xml");
+    std::string pathBIN(pathXML);
+    return pathBIN.replace(pathBIN.size() - 3, 3, "bin");
```

Comment:
may not work with `.BIN` in a case-sensitive OS/FS. Still, not sure if it is the case